### ISSUE TITLE
Setting -b -t -p

### ISSUE THREAD

**Issue by ghost** [brichard19/BitCrack#308]
Been testing and getting around 107 MKey/S on a GTX 1650 with -b 180 -t 128 -p 1088.
Is there a way to figure out the correct settings instead of trial/error? Been checking around if I can find the correct values, but so far no luck.

Nvidia system information says CUDA-cores 896 and memory interface 128 bits.

Currently running cl-version as I don't get cu-version to work.

**Comment by MisterTeo** [comment]
Try to compile this version https://github.com/djarumlights/BitCrack/releases
I have the same card an I use cuBitCrack --compression BOTH -b 64 -t 128 -p 3200 and 37 000 keys in -i file. All tht and I get 111.85 MKey/s...

**Comment by ghost** [comment]
Thanks
Okay, changed my settings to yours and getting between 112-116MKeys/S now with 48,5K keys.
The first address I had in the file was a known address, with a to me known hex. The clBitCrack triggered  immediately and I got
the bitcoinaddress, hexadecimal and public key in my designated .txt. Converted to compressed WIF and can confirm it was valid, even though empty. See 18ZMbwUFLMHoZBbfpCjUJQTCMCbktshgpe (it wasn't me that emptied it).

**Comment by ghost** [comment]
If someone has a way to calculate best -b -t -p please post it here.

**Comment by ghost** [comment]
for cuda version only: -b (number of multiprocessors*8) -t (128|256|512) -p (1024|2048)
1050Ti runs best with these settings: cuBitCrack -b 48 -t 128 -p 1024

**Comment by mrantonion** [comment]
Dear friends. What for are you burning an electricity with 37K addresses in your pocket? ))) The probability to find smth is 0,00000000000000%. Spend a little bit time to add all filled addresses. As for me I'm scanning near 13M BTC addresses with balance (min balance is 100K satoshi). The speed on 1080Ti is 160 Mkeys/s.

**Comment by marssystems** [comment]
Have you found anything yet?

**Comment by mrantonion** [comment]
Of course no! )) That's impossible without some algorithm in your mind. People just can't image 2^256 number. So they try and try to find smth, later the enthusiasm finishes. ))

**Comment by marssystems** [comment]
Yes I agree. The algorithm in my mind is "pure luck" and the bonus is that my house is nice and warm!

**Comment by OrbitalCombustion** [comment]
If you peruse the BitcoinTalk forum conversations pertaining to this, you can give yourself a big head start. Regarding #64 for the BTC Puzzle (which really isn't a puzzle but rather a demonstration of the vastness of the private key space), you can use a hex converter to start in the middle or any custom range you want. Its more likely that the beginning and ending of the #64 range has been exhausted by all manner of amateur BitCrackers. If anyone needs assistance please let me know, I have a bunch of resources regarding this amazing hidden treasure hunt.

**Comment by OrbitalCombustion** [comment]
Just to give you an idea of how long it would take to just mine the private key on puzzle #64 on say an ASUS ROG STRIX GTX 1080 8GB OC, averaging ~115 MKey/s, here is the math: 

Hex range: 8000000000000000:ffffffffffffffff
Respective decimal range: 9223372036854775808:18446744073709551615

Total number of keys in range: 9,223,372,036,854,775,807
Number of seconds required to crack: 80,203,235,103
Number of minutes required to crack: 1,336,720,585
Number of hours required to crack: 22,278,676
Number of days required to crack: 928,278
Number of years required to crack: 2,543

The guys who are cracking these higher register numbers have at their disposal supercomputers and/or elite programming skills for biginteger math. You can find all the associated math on BitcoinTalk forum, I can send link, message me. At the end of the day much of this is based on your ability to think outside the box in terms of hex ranges, in addition to sheer luck/chance. Good luck to all!

**Comment by mrantonion** [comment]
I doubt that 1080 gives a 115 Mk/s speed. I'm using 1080Ti and it gives 350 Mk/s. The speed highly depends on the quantity of  simultaneously searching addresses. If you are searching only for 1 address in the defined range the speed is much higher than expected.

**Comment by OrbitalCombustion** [comment]
It is 115 MKey/s, and yes I am searching the known #64 address with a custom hex range. The reason I quoted that is because that is the card I have and the speeds I am getting.

**Comment by OrbitalCombustion** [comment]
I haven't even modded the parameters yet; so I suspect I will be able to get a higher MKey/s once I tinker a bit more. Scanning multiple addresses against a large hex range seems futile, but deserving of a try nonetheless. Since all of the addresses for the 'puzzle' have been identified and their respective hex ranges, it makes more sense to experiment within those parameters.

**Comment by OrbitalCombustion** [comment]
For instance, using the equation 2^0 < n < 2^1, we can see that the private keys are random numbers within the parameters of that equation. Here is an example going to puzzle #14.

Hex(Dec):Hex(Dec) - Known private key: Hex(Dec)

1. 1(1):1(1) - 1 (1)
2. 2(2):3(3) - 3 (3)
3. 4(4):7(7) - 7 (7)
4. 8(8):f(15) - 8 (8)
5. 10(16):1f(31) - 15 (21)
6. 20(32):3f(63) - 31 (49)
7. 40(64):7f(127) - 4c (76)
8. 80(128):ff(255) - e0 (224)
9. 100(256):1ff(511) - 1d3 (467)
10. 200(512):3ff(1023) - 202 (514)
11. 400(1024):7ff(2047) - 483 (1155)
12. 800(2048):fff(4095) - a7b (2683)
13. 1000(4096):1fff(8191) - 1460 (5216)
14. 2000(8192):3fff(16383) - 2930 (10544)

**Comment by OrbitalCombustion** [comment]
@mrantonion Also regarding your comment about algorithms, that topic has been beaten like the dead horse it is by people much more involved in this than you or me. The 'x' in '2^n < x < 2^n+1' is random, and has nothing whatsoever to do with an algorithm, or sequence or anything predictable. That literally defeats the purpose of why it was deposited. As I advised earlier, do yourself a favor and read through the couple hundred pages on BitcoinTalk forum regarding this.

**Comment by mrantonion** [comment]
I don't want to boast but the regularity in puzzle's addresses does exist. But it's not my interest to divulge it even despite the fact that the knowing of this regularity is much useful (with my weak GPU power).

**Comment by OrbitalCombustion** [comment]
@mrantonion Dude you are fooling yourself if you think you found any 'regularity' or 'algorithm' or anything predictable about the addresses. The creator of the puzzle literally posted on BitcoinTalk stating that the private key selection was completely random based on this equation: 2^n < x < 2^n+1, where x = random 64 digit integer. You're not boasting shit because you're full of it. Your best bet with any of this is using bigstep/bigint math. If you think any differently you havent done enough research on this puzzle or are blinding yourself by trying to find an algorithm when people much smarter than you or I mathematically have already disproven this. Plus its obvious. Trust me, the first thing I looked for was some kind of regularity in the sequence, however if that were true it would have been solved long ago and all the BTC would have been swept. Hopefully you use your time more wisely going forward.

**Comment by bikkesbakker** [comment]
> I doubt that 1080 gives a 115 Mk/s speed. I'm using 1080Ti and it gives 350 Mk/s. The speed highly depends on the quantity of simultaneously searching addresses. If you are searching only for 1 address in the defined range the speed is much higher than expected.

Hey, what parameters do you have configured ? can you remember ?

**Comment by ejrhe** [comment]
do any body wants to about me i am getting 445 MKey/s speed
